wallet-next/src/bin/pwalletd.rs

```diff
@@ -1,22 +1,210 @@
+        #[structopt(short, long, default_value = "26666")]
+        oblivious_query_port: u16,
+        /// The port to use to speak to pd's thin wallet server.
+        #[structopt(short, long, default_value = "26667")]
+        specific_query_port: u16,
+    },
+}
 #[tokio::main]
 async fn main() -> Result<()> {
-    let pool = SqlitePool::connect(&env::var("DATABASE_URL")?).await?;
-    let storage = Storage::new(pool);
+    tracing_subscriber::fmt::init();
+    let opt = Opt::from_args();
+
+    match opt.cmd {
+        Command::Start {
+            sqlite_path,
+            host,
+            grpc_port,
+            node,
+            oblivious_query_port,
+            specific_query_port,
+        } => {
+            tracing::info!(
+                ?sqlite_path,
+                ?host,
+                ?grpc_port,
+                ?node,
+                ?oblivious_query_port,
+                ?specific_query_port,
+                "starting pwalletd"
+            );
+
+            let mut client =
+                ObliviousQueryClient::connect(format!("http://{}:{}", node, oblivious_query_port))
+                    .await
+                    .context("Unable to connect to oblivious query server")?;
+
+            let pool = SqlitePool::connect(&env::var("DATABASE_URL")?).await?;
+            let storage = Storage::new(pool);
+            storage.migrate().await?;
+
+            // TODO: select a custody service here to provide the wallet data source and update local sqlite storage as needed
+
+            // Fetch chain params if necessary so we can .expect() on them.
+            if storage.chain_params().await.is_none() {
+                chain_params(&client, &mut storage).await?;
+            }
+
+            // Always sync pwalletd on startup.
+            sync(&client, &mut storage).await?;
+            // Retrieve asset list
+            assets(&client, &mut storage).await?;
+
+            // TODO: this is just a sqlite usage stub
+
+            let row = storage.insert_table().await?;
+            let x = storage.read_table().await?;
+
+            println!(
+                "Hello, pwalletd! I got stuff from sqlite: row {} value {}",
+                row, x
+            );
+
+            // TODO: start gRPC service and respond to command requests
+            // let wallet_server = tokio::spawn(
+            //     Server::builder()
+            //         .trace_fn(|req| match remote_addr(req) {
+            //             Some(remote_addr) => tracing::error_span!("wallet_query", ?remote_addr),
+            //             None => tracing::error_span!("wallet_query"),
+            //         })
+            //         .add_service(WalletServer::new(storage.clone()))
+            //         .serve(
+            //             format!("{}:{}", host, wallet_query_port)
+            //                 .parse()
+            //                 .expect("this is a valid address"),
+            //         ),
+            // );
```

Comment:
This logic isn't going to be what we really want (we'll need to start and manage tasks to run the wallet service), so let's replace the whole block with an `unimplemented!()` and fill it in later -- there shouldn't be much code in the `pwalletd` binary itself anyways, since that's supposed to just be another way to run the common wallet service code that could also be in-process for `pcli`.